go/cmd/dolt/commands/log_graph.go

```diff
@@ -0,0 +1,428 @@
+// Copyright 2019 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"fmt"
+	"math"
+	"strings"
+
+	"github.com/dolthub/dolt/go/cmd/dolt/cli"
+	"github.com/dolthub/dolt/go/libraries/utils/argparser"
+	"github.com/dolthub/dolt/go/store/util/outputpager"
+)
+
+type CommitInfoWithChildren struct {
+	Commit           CommitInfo
+	Children         []string
+	X                int
+	Y                int
+	formattedMessage []string
+}
+
+var branchColors = []string{
+	"\033[31m", // Red
+	"\033[32m", // Green
+	"\033[34m", // Blue
+	"\033[35m", // Magenta
+	"\033[36m", // Cyan
+	"\033[39m", // Default
+}
+
+type BranchPathType struct {
+	Start int
+	End   int
+}
+
+// get the children of commits, and initialize the x and y coordinates of the commits
+func mapCommitsWithChildrenAndPosition(commits []CommitInfo) []*CommitInfoWithChildren {
+	childrenMap := make(map[string][]string)
+	for _, commit := range commits {
+		for _, parent := range commit.parentHashes {
+			childrenMap[parent] = append(childrenMap[parent], commit.commitHash)
+		}
+	}
+
+	var commitsWithChildren []*CommitInfoWithChildren
+	for y, commit := range commits {
+		commitsWithChildren = append(commitsWithChildren, &CommitInfoWithChildren{
+			Commit:   commit,
+			Children: childrenMap[commit.commitHash],
+			X:        -1,
+			// the y coordinate of the commit is initialized to the index of the commit as the commits are sorted
+			Y: y,
+		})
+	}
+
+	return commitsWithChildren
+}
+
+// wrap the commit message in a constrained width to better align the commit message with the graph
+func wrapTextOnWidth(text string, width int) []string {
+	lines := strings.Split(text, "\n")
+	totalRows := 0
+	wrappedLines := make([]string, 0)
+
+	for _, line := range lines {
+		words := strings.Fields(line)
+		currentLine := ""
+		for _, word := range words {
+			if len(currentLine)+len(word)+1 > width {
+				wrappedLines = append(wrappedLines, currentLine)
+				totalRows++
+				currentLine = word
+			} else {
+				if currentLine != "" {
+					currentLine += " "
+				}
+				currentLine += word
+			}
+		}
+		if currentLine != "" {
+			wrappedLines = append(wrappedLines, currentLine)
+			totalRows++
+		}
+	}
+	return wrappedLines
+}
+
+func min(values ...int) int {
```

Comment:
There is a buildin function by the same name which takes only two args. Rename this to `minVal` to avoid confusion.